src/main/java/graphql/schema/DataFetchingEnvironmentImpl.java

```diff
@@ -210,6 +212,18 @@ public ExecutionStepInfo getExecutionStepInfo() {
         return dataLoaderRegistry.getDataLoader(dataLoaderName);
     }
 
+    @Override
+    public DataLoaderChain getDataLoaderChain() {
+        if (dataLoaderChain == null) {
+            synchronized (this) {
+                if (dataLoaderChain == null) {
+                    dataLoaderChain = new DataLoaderChain(this);
+                }
+            }
+        }
+        return dataLoaderChain;
```

Comment:
```suggestion
        DataLoaderChain chain = this.dataLoaderChain;
        if (chain == null) {
            synchronized (this) {
                if ((chain = this.dataLoaderChain) == null) {
                    chain = this.dataLoaderChain = new DataLoaderChain(this);
                }
            }
        }
        return chain;
```

Reply:
Results in fewer volatile reads

Reply:
We should not use synchronized in a virtual threads world if we can.  We should move this code into a LockKit world

Reply:
Good shout - it's "safe"(r) in JDK >= 24 but definitely don't want to make that assumption.